src/app/components/wizard/wizard.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, ContentChildren, Input, QueryList } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Router } from '@angular/router';
+import { WizardStepDirective } from 'src/app/directives/wizard-step.directive';
+
+@Component({
+  selector: 'app-wizard',
+  templateUrl: './wizard.component.html',
+  styleUrls: ['./wizard.component.scss'],
+})
+export class WizardComponent {
+  @ContentChildren(WizardStepDirective)
+  stepsQueryList!: QueryList<WizardStepDirective>;
+
+  steps: WizardStepDirective[] = [];
+  stepIndex = 0;
+
+  @Input() shippingForm!: FormGroup;
+  @Input() paymentForm!: FormGroup;
+
+  constructor(private router: Router) {}
+
+  isLastStep(): boolean {
+    return this.stepIndex + 1 === this.steps.length;
+  }
```

Comment:
I recommend having this as a `getter`. `get isLastStep() ...` to clearly define calculated properties